ingest/expression_files/expression_files.py

```diff
@@ -64,6 +64,41 @@ def set_dataArray(self):
         Each expression file will have its own implementation of setting the
         DataArray with expression data."""
 
+    @staticmethod
+    def has_unique_cells(cell_names: List, study_id, client):
+        """Checks cell names against database to confirm matrix contains unique
+            cell names
+
+         Parameters:
+            cell_names (List[str]): List of cell names in matrix
+            study_id (ObjectId): The study id the cell names belong to
+            client : MongoDB client
+        """
+        COLLECTION_NAME = "data_arrays"
+        query = {
+            "$and": [
+                {"linear_data_type": "Study"},
```

Comment:
Not strictly necessary, but you could add `{"linear_data_id": study_id}`.

Reply:
Im adding this suggestion in another PR.